codalab/model/tables.py

```diff
@@ -74,6 +74,20 @@
   sqlite_autoincrement=True,
 )
 
+# Stores information about the files, directories and links stored in the
+# bundle.
+bundle_contents_index = Table(
+  'bundle_contents_index',
+  db_metadata,
+  Column('bundle_uuid', String(63), ForeignKey(bundle.c.uuid), nullable=False),
+  Column('path', Text, nullable=False),
+  Column('type', String(63), nullable=False),
+  Column('size', Integer, nullable=False),
+  Column('perm', Integer, nullable=False),
+  Column('link', Text, nullable=True),
```

Comment:
Document what each of these fields are (for 'type', give the possible values)

General comment: Now that 'type' is more offiical/permanent, just like for permissions, let's define constants FILETYPE_FILE, FILETYPE_DIRECTORY, and FILETYPE_LINK instead of using 'file', 'directory', and 'link' throughout the code.